clusterlib/scheduler.py

```diff
@@ -53,10 +53,15 @@ def is_exe(fpath):
 
 def _get_backend(backend="auto"):
     """Detect the backend to use based on the commands present in the PATH"""
-    backend_commands = (
-        ('slurm', 'squeue'),
-        ('sge', 'qstat'),
-    )
+    # To detect if the backend is available, it's better to detect the presence
+    # of administrator tools over launching (sbatch, qsub) or job status
+    # (qstat, squeue) commands since a proxy might be provided by
+    # the cluster distribution such as the rock roll cluster distribution.
+    backend_commands = {
+        'slurm': 'scontrol',
+        'sge': 'qmod',
+    }
```

Comment:
The ordering of a Python dict is not deterministic. The use of a tuple of pairs as intended to keep the ordering of the tests deterministic to select slurm over sge when both are available on the host.